### DISCUSSION TITLE
Triggered device that caused the alarm

### DISCUSSION THREAD

**Discussion by iptvcld** [tsightler/ring-mqtt#893] [Support Q&A]
Hello; does anyone know if there is a sensor that captures the alarm sensor which caused the alarm to trigger?

Thanks,

**Comment by tsightler** [comment]
Duplicate of https://github.com/tsightler/ring-mqtt/discussions/877.  Same answer.

**Reply by iptvcld** [reply]
Thanks, I am using a template sensor right now that has the group of sensors and as soon as one of them changes to On, it stores that in a var which i later use to call and send a SMS with the stored var to indicate what caused the alarm.  (trying to copy the email Ring sends out as well).  But issue i sometimes face is during a false alarm when i open and close the door very quick, it triggers the alarm but since it was too quick, it does not store the triggered sensor in the var and so i dont know what caused it.

**Reply by tsightler** [reply]
I think you could probably work around this by looking for any sensor where last_changed was in the last few seconds.

Unfortunately, I don't currently know any way to easily grab this information.  The push notification that Ring sends out is generated by the server based on information from the hub which is triggered locally on the Z-wave network.  it is not directly exposed via the API, at least not via a standard state change.  Maybe it could be queried from the event history and it is probably available on the message bus, but I lack the time to spend deciphering the entire messaging protocol.

The new style push notifications seem promising, but alarm notifications don't use the new format yet as far as I can tell anyway.
